engine/setup/src/include/general.cpp

```diff
@@ -104,7 +104,7 @@ char *ptr_copy(const char *string)
 #ifdef __STDC_LIB_EXT1__
     strcpy_s(alloc, buf_size, string);
 #else
-    strncpy(alloc, string, buf_size);
+    strncpy_s(alloc, buf_size, string, buf_size);
```

Comment:
Hmm. There may be a hidden compatibility issue here or two, to revisit at some point. But probably not a big deal. The code seems to be building on all presently supported platforms, at any rate.

Reply:
Just stick to `strncpy`; `strncpy_s` is an MSVCism and doesn't really buy you anything over `strncpy`; worse case we have to add a header to rename `_strncpy` to `strncpy` or something like that. But let's keep it platform agnostic as much as possible.